kotlin-generator/src/helpers.ts

```diff
@@ -0,0 +1,240 @@
+import {
+    ArrayType,
+    Base64PrimitiveType,
+    BigIntPrimitiveType,
+    BoolPrimitiveType,
+    BytesPrimitiveType,
+    CnpjPrimitiveType,
+    CpfPrimitiveType,
+    DatePrimitiveType,
+    DateTimePrimitiveType,
+    EmailPrimitiveType,
+    EnumType,
+    FloatPrimitiveType,
+    HexPrimitiveType,
+    IntPrimitiveType,
+    JsonPrimitiveType,
+    MoneyPrimitiveType,
+    OptionalType,
+    StringPrimitiveType,
+    StructType,
+    Type,
+    TypeReference,
+    UIntPrimitiveType,
+    UrlPrimitiveType,
+    UuidPrimitiveType,
+    VoidPrimitiveType,
+    XmlPrimitiveType,
+} from "@sdkgen/parser";
+
+export function generateEnum(type: EnumType) {
+    return `enum class ${type.name} { ${type.values.map(x => mangle(x.value)).join(", ")} }\n`;
+}
+
+export function generateClass(type: StructType) {
+    return `data class ${type.name}(\n${type.fields
+        .map(field => {
+            let fieldDesc = "";
+            if (field.type.constructor === DatePrimitiveType) {
+                fieldDesc += "        @JsonAdapter(DateAdapter::class)\n";
+            } else if (field.type.constructor === DateTimePrimitiveType) {
+                fieldDesc += "        @JsonAdapter(DateTimeAdapter::class)\n";
+            }
+            fieldDesc += `        val ${mangle(field.name)}: ${generateKotlinTypeName(field.type)}`;
+            return fieldDesc;
```

Comment:
if something has type `date?` it won't get the annotation.

Reply:
I didn't undestand
The code is this: 
```
if (field.type.constructor === DatePrimitiveType) {
                fieldDesc += "        @JsonAdapter(DateAdapter::class)\n";
            } else if (field.type.constructor === DateTimePrimitiveType) {
                fieldDesc += "        @JsonAdapter(DateTimeAdapter::class)\n";
            }
```
If the field is a Date, it will get the TypeAdpater annotation correctly 🤔

Reply:
Because the constructor will be OptionalType

Reply:
`date?` is a OptionalType whose `.base` is a DatePrimitiveType. And you can have things like `date???[]?[]`